pennylane/operation.py

```diff
@@ -799,6 +799,98 @@ def eigvals(self):
+    defined for all channels:
+
+    * :attr:`~.Operator.num_params`
+    * :attr:`~.Operator.num_wires`
+    * :attr:`~.Operator.par_domain`
+
+    The following two class attributes are optional, but in most cases
+    should be clearly defined to avoid unexpected behavior during
+    differentiation.
+
+    * :attr:`~.Operation.grad_method`
+    * :attr:`~.Operation.grad_recipe`
+
+    Args:
+        params (tuple[float, int, array, Variable]): operation parameters
+
+    Keyword Args:
+        wires (Sequence[int]): Subsystems the channel acts on. If not given, args[-1]
+            is interpreted as wires.
+        do_queue (bool): Indicates whether the operation should be
+            immediately pushed into a :class:`BaseQNode` circuit queue.
+            This flag is useful if there is some reason to run an Operation
+            outside of a BaseQNode context.
+    """
+    # pylint: disable=abstract-method
+
+    @classmethod
+    def _kraus_matrices(cls, *params):
+        """Kraus matrices representing a quantum channel, specified in
+        the computational basis.
+
+        This is a class method that should be defined for all
+        new channels. It returns the Kraus matrices representing
+        the channel in the computational basis.
+
+        This private method allows matrices to be computed
+        directly without instantiating the channel first.
+
+        **Example:**
+
+        >>> qml.AmplitudeDamping._kraus_matrices(0.1)
+        >>> [array([[1.       , 0.       ],
+        [0.       , 0.9486833]]), array([[0.        , 0.31622777],
+        [0.        , 0.        ]])]
+
+        To return the Kraus matrices of an *instantiated* channel,
+        please use the :attr:`~.Operator.kraus_matrices` property instead.
+
+        Returns:
+            list(array): list of Kraus matrices
+        """
+        raise NotImplementedError
+
+    @property
+    def kraus_matrices(self):
+        r"""Kraus matrices of an instantiated channel
+        in the computational basis.
+
+        ** Example: **
+
+        >>> U = qml.AmplitudeDamping(0.1, wires=1)
+        >>> U.kraus_matrices
+        >>> [array([[1.       , 0.       ],
+        [0.       , 0.9486833]]), array([[0.        , 0.31622777],
+        [0.        , 0.        ]])]
+
+        Returns:
+            list(array): list of Kraus matrices
+        """
+        return self._kraus_matrices(*self.parameters)
+
+    def __init__(self, *params, wires=None, do_queue=True):
+
+        self._inverse = False
+
+        # check the grad_method validity
+        if self.par_domain == "R":
+            assert self.grad_method in (
+                None,
+                "F",
+            ), "Analytic gradients can not be used for quantum channels, presently."
+
+        # check the grad_recipe validity
+        if self.grad_method == "F":
+            assert self.grad_recipe is None, "Gradient recipe are not be defined for quantum channels, presently."
```

Comment:
Same comment here regarding validation.

However, I think this check can be removed; if `grad_method="F"`, then the `grad_method` doesn't apply anyway.